backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/TypeOperatorLowering.kt

```diff
@@ -75,6 +63,34 @@ private class TypeOperatorTransformer(val context: CommonBackendContext, val fun
         }
     }
 
+    fun IrBuilderWithScope.irCast(argument: IrExpression, type: IrType, argumentIsNotNull: Boolean = false) = irBlock {
+        val value = irTemporary(argument)
+        +irIfThenElse(
+                type,
+                condition = irInstanceOf(value, type, argumentIsNotNull),
+                thenPart = irImplicitCast(irGet(value), type),
```

Comment:
Note that implicit cast is handled by autoboxing through `Any?` type, it can box argument and unbox result. Please make sure this new cast implementation doesn't affect performance.